homeassistant/components/vizio/media_player.py

```diff
@@ -135,6 +133,7 @@ def __init__(self, host, token, name, volume_step, device_type):
         self._supported_commands = SUPPORTED_COMMANDS[device_type]
         self._device = Vizio(DEVICE_ID, host, DEFAULT_NAME, token, device_type)
         self._max_volume = float(self._device.get_max_volume())
+        self._unique_id = f"{device_type}-{self._device.get_esn()}"
```

Comment:
`self._unique_id` should be `None` if the ESN could not be found.

Reply:
This is why I should be asleep right now... Thanks